src/app/pages/OnDemandRadioPage/index.jsx

```diff
@@ -1,14 +1,53 @@
 import React, { useContext } from 'react';
-import { string, shape } from 'prop-types';
+import { string, number, shape } from 'prop-types';
 import styled from 'styled-components';
 import MetadataContainer from '../../containers/Metadata';
 import Grid, { GelPageGrid } from '#app/components/Grid';
 import { ServiceContext } from '../../contexts/ServiceContext';
-
 import HeadingBlock from '#containers/RadioPageBlocks/Blocks/Heading';
 import ParagraphBlock from '#containers/RadioPageBlocks/Blocks/Paragraph';
+import AudioPlayerBlock from '#containers/RadioPageBlocks/Blocks/AudioPlayer';
 
 const SKIP_LINK_ANCHOR_ID = 'content';
+const EPISODE_IS_AVAILABLE = 'is-available';
+const EPISODE_IS_EXPIRED = 'is-expired';
+const EPISODE_IS_NOT_YET_AVAILABLE = 'is-not-yet-available';
+
+const getTimeTicks = () => new Date().getTime() * 10000;
+
+const getEpisodeAvailability = (from, until) => {
+  const timeNow = getTimeTicks();
+  if (timeNow > until) return EPISODE_IS_EXPIRED;
```

Comment:
I believe the behavior when an episode has expired is that it does not get returned at all, ie, the `versions` in the payload is an empty array

See the live ares response for `pashto/bbc_pashto_radio/w172x8nvf4bchz5` for an example

Not sure if the code handles this correctly

Reply:
is this the same for when an episode is not yet available?

Reply:
I've seen payloads with "availableFrom" in the future so I'd guess they always get returned.  Just a guess though - probably worth asking Jane or AReS what we should be expecting